File: Demos/Datasnap/Server/proxy/java_blackberry/com/embarcadero/javablackberry/DBXException.java
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

package com.embarcadero.javablackberry;

/**
 * Exception descendant to raise error in proxy
 *
 */

public class DBXException extends Exception {

	private Throwable Internal;
	
	public DBXException(String value) {
		super(value);
	}
	
	public DBXException(Throwable value) {
		super(value.getMessage());
		Internal = value;
	}

	public Throwable getInternal()
	{
		return Internal;
	}	
	
	public String getMessage()
	{
		String res;
		if ((res = super.getMessage()) == null)
			if (getInternal() != null)
				res = getInternal().getMessage();
		return res;
	}
	
	private static final long serialVersionUID = 154857855606412539L;
}

File: Demos/Datasnap/Server/proxy/csharp_silverlight/DBXException.cs
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

using System;

namespace Embarcadero.Datasnap.WindowsPhone7
{
    
   /**
   * Exception descendant to raise error in proxy
   *
   */
 
    class DBXException : Exception
    {
        public DBXException(String value): base(value) {
	    }

	    /**
	     * 
	     */
	    private const long serialVersionUID = 154857855606412539L;

    }
}
